ene/models.py

```diff
@@ -0,0 +1,243 @@
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""This module handles data and data access."""
+from pathlib import Path
+import re
+from enum import Enum
+from peewee import Model, SqliteDatabase, TextField, IntegerField, ForeignKeyField
+
+db = SqliteDatabase(None)
+
+
+class EneDatabase:
+    def __init__(self, path):
+        """
+        Initializes the Ene database, creating any tables as needed
+
+        Args:
+            path:
+                The path where the Sqlite database file resides
+        """
+        self.database = db
+        db.init(path)
+        db.connect()
+        db.create_tables([ShowModel, EpisodeModel])
+
+
+def table_name(table):
+    """
+    Derives the database table name from the Peewee Model class
+
+    Args:
+        table:
+            The model object referencing a table in the Ene database
+
+    Returns:
+        The name of the table it represents (strips "Model" from the name)
+    """
+    return table.__name__.replace('Model', '')
+
+
+class BaseModel(Model):
+    """
+    Base model for all Ene models, provides base behaviour of the database
+    and the table naming function
+    """
+    class Meta:
+        """ Meta class for BaseModel which defines settings"""
+        database = db
+        table_function = table_name
+
+
+class Show:
+    """
+    Class containing information about a single show
+    """
+    def __init__(self, title, show_id=None, list_id=None, episodes: set = None, key=None):
+        self.title = title
+        self.show_id = show_id
+        self.list_id = list_id
+        if episodes is None:
+            self.episodes = set()
+        else:
+            self.episodes = episodes
+        self.key = key
+
+    def add_episode(self, episode):
+        """
+        Adds a single episode to the shows set of episodes
+
+        Args:
+            episode: The episode to add
+        """
+        self.episodes.add(episode)
+
+    def __len__(self):
+        """
+        The number of episodes in the show
+
+        Returns:
+            An integer representing the length of the episodes set
+        """
+        return len(self.episodes)
+
+
+class ShowModel(BaseModel):
+    """
+    Model representing a Show in the database
+    """
+    title = TextField()
+    anilist_show_id = IntegerField(null=True)
+    list_id = IntegerField(null=True)
+
+    @staticmethod
```

Comment:
nitpicking, but this should be a class method and return `cls(title=show.title ...` because inheritance behaviors, same goes for EpisodeModel